cg/apps/sequencing_metrics_parser/parsers/bcl2fastq.py

```diff
@@ -88,17 +89,16 @@ def discard_index_sequence(sample_id_with_index: str) -> str:
 def get_bcl2fastq_stats_paths(demultiplex_result_directory: Path) -> List[Path]:
     """Find metrics files in Bcl2fastq demultiplex result directory."""
     stats_json_paths = []
-    pattern = re.compile(r"l\d+t\d+")
-
-    for subdir in os.listdir(demultiplex_result_directory):
-        if pattern.match(subdir):
-            stats_json_path = (
-                demultiplex_result_directory
-                / subdir
-                / BCL2FASTQ_METRICS_DIRECTORY_NAME
-                / BCL2FASTQ_METRICS_FILE_NAME
-            )
-            if stats_json_path.is_file():
+
+    for root, dirs, files in os.walk(demultiplex_result_directory):
+        if BCL2FASTQ_METRICS_FILE_NAME in files:
+            stats_json_path = Path(root, BCL2FASTQ_METRICS_FILE_NAME)
+            if root.endswith(BCL2FASTQ_METRICS_DIRECTORY_NAME):
                 stats_json_paths.append(stats_json_path)
 
+    if not stats_json_paths:
+        raise FileNotFoundError(
```

Comment:
Did we settle on exceptions being included in the doc string?

Reply:
I think it should be included. Adding it 👍